### PR TITLE
docs: Document how to fetch additional branches

### PR DESCRIPTION
**Pull request by stephenfin**
Tools like reno depend on a complete git history. Simply unshallowing a clone is not sufficient, as we specifically fetch from the branch or tag that we are building docs for, e.g.:

    git fetch origin --force --prune --prune-tags --depth 50 refs/heads/main:refs/remotes/origin/main

Add documentation describing how you can correct this if necessary.

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--10795.org.readthedocs.build/en/10795/

- Developer's documentation (`dev`): https://dev--10795.org.readthedocs.build/en/10795/

### CODE REVIEW

File: docs/user/build-customization.rst [hunk 1]
```diff
@@ -104,6 +105,22 @@ To avoid this, it's possible to unshallow the :program:`git clone`:
...
+
+   version: 2
+   build:
+     os: "ubuntu-20.04"
+     tools:
+       python: "3.10"
+     jobs:
+       post_checkout:
+         - git fetch --unshallow || true
+         - git config remote.origin.fetch '+refs/heads/*:refs/remotes/origin/*' || true
```
Comment (ericholscher) [inline docs/user/build-customization.rst:121]: Why is this line required? Seems pretty specific to a use case, and something I'd expect git to do by default?

**Review by ericholscher — COMMENTED** [review]
Thanks for the contribution. If there's a specific tool that this is useful for, it's probably worth mentioning in the PR as well.

File: docs/user/build-customization.rst (hunk 1 above)
Comment (stephenfin) [reply docs/user/build-customization.rst:121]: This is necessary because we're configuring git to only fetch a single branch in this command (which you'll see a variant of in all builds):

```
git fetch origin --force --prune --prune-tags --depth 50 refs/heads/main:refs/remotes/origin/main
```

If you shallow clone a repo then run the above locally and look at `.git/config`, you'll see something like the following:

```
[remote "origin"]
fetch = +refs/heads/main:refs/remotes/origin/main
```

Which means "map the `main` ref locally to the `main` remote ref". Running `git fetch` will always result in just this branch being fetched. To fix this, you need to tell git to fetch all branches and map them to the same name locally, which is what we're doing here.

Comment (stephenfin) [reply docs/user/build-customization.rst:121]: Oh, and as I mentioned in the blurb beforehand, you'll need to do this if you have a tool that needs access to other branches rather than just other tags.

**Review by ericholscher — APPROVED** [review]
